modules/mono/glue/GodotSharp/GodotSharp/Core/Quaternion.cs

```diff
@@ -558,26 +558,44 @@ public Quaternion(Vector3 axis, real_t angle)
 
         public Quaternion(Vector3 arcFrom, Vector3 arcTo)
         {
-            Vector3 c = arcFrom.Cross(arcTo);
-            real_t d = arcFrom.Dot(arcTo);
-
-            if (d < -1.0f + Mathf.Epsilon)
+#if DEBUG
+            if (arcFrom.IsZeroApprox() || arcTo.IsZeroApprox())
             {
-                X = 0f;
-                Y = 1f;
-                Z = 0f;
-                W = 0f;
+                throw new ArgumentException("The vectors must not be zero.");
+            }
+#endif
+#if REAL_T_IS_DOUBLE
+            const real_t AlmostOne = 0.999999999999999;
+#else
+            const real_t AlmostOne = 0.99999975f;
+#endif
+            Vector3 n0 = arcFrom.Normalized();
+            Vector3 n1 = arcTo.Normalized();
+            real_t d = n0.Dot(n1);
+            if (Mathf.Abs(d) > AlmostOne)
+            {
+                if (d >= 0)
+                {
+                    return; // Vectors are same.
+                }
+                Vector3 axis = n0.GetAnyPerpendicular();
+                X = axis.X;
+                Y = axis.Y;
+                Z = axis.Z;
+                W = 0;
             }
             else
             {
-                real_t s = Mathf.Sqrt((1.0f + d) * 2.0f);
-                real_t rs = 1.0f / s;
+                Vector3 c = n0.Cross(n1);
+                real_t s = Mathf.Sqrt((1f + d) * 2f);
+                real_t rs = 1f / s;
```

Comment:
The existing code uses `1.0f`, and I prefer `1.0f`, so this shouldn't be changed unless there is agreement somewhere that `1f` is preferred.